src/OAuth2.js

```diff
@@ -59,11 +59,13 @@ function createService(serviceName) {
 /**
  * Returns the redirect URI that will be used for a given script. Often this URI
  * needs to be entered into a configuration screen of your OAuth provider.
- * @param {string} scriptId The script ID of your script, which can be found in
- *     the Script Editor UI under "File > Project properties".
+ * @param {string} [optScriptId] The script ID of your script, which can be
+ *     found in the Script Editor UI under "File > Project properties". Defaults
+ *     to the script ID of the script being executed.
  * @return {string} The redirect URI.
  */
-function getRedirectUri(scriptId) {
+function getRedirectUri(optScriptId) {
+  var scriptId = optScriptId || eval('Script' + 'App').getScriptId();
```

Comment:
The eval('Script' + 'App') call here is weird -- why necessary to eval and why the concatenation of constants? Shouldn't this just be ScriptApp.getScriptId()?

Reply:
BTW - If this was an attempt to thwart oauth scope detection, it looks like calling ScriptApp.getScriptId() directly doesn't actually trigger an auth requirement and is safe to use.

Reply:
It was a workaround to scope issues, but it looks like the problem it was working around was fixed in mid-2017, so no longer needed. Removed.